### ISSUE TITLE
applying seisbench to 1C data

### ISSUE THREAD

**Issue by dttrugman** [seisbench/seisbench#123]
First, thanks for making this awesome package available to the public! It is very well done and a great resource to the community.

Now, a question. I'm interested in trying to apply some of the pre-trained models to vertical component (1C) datasets. While most models seem to have been optimized for 3C waveforms, some of the native packages like EQT seem to support this in some capacity and perform pretty well. [Aside: it is not clear in their documentation how this is done, but it seems to work]. In terms of seisbench, is there an easy way to reformulate a model to be compatible with single-component waveforms? Or is the only hope to have to re-train from scratch?

Likely the ideal long-term solution may be transfer learning (taking a pre-trained model, reshaping, and retraining), but I am a bit stuck getting started...

Any advice or suggestions would be most welcome; hopefully this is not a dumb question (I didn't find any similar ones in the Issue archive).

**Comment by yetinam** [comment]
Hi Daniel,

thanks for the kind words on SeisBench! Regarding your question, there is a more syntactic and a more semantic answer. Syntactically, you can just use `model.annotate(stream, strict=False)` or the same with `classify`. This works for every model and will just feed zeros to the model for every missing component of the stream.

Now clearly the question is if this make sense semantically. From what I've seen so far, (our implementations of) EQTransformer, PhaseNet and GPD all give quite decent results even if input channels are missing. As you mentioned, EQTransformer is actually trained for this case. This is done (both in the original and for our versions) through data augmentation. We implemented this as a module called [ChannelDropout](https://github.com/seisbench/seisbench/blob/ca0c2966ff30fbd009378dc0dfda60a285aa1efd/seisbench/generate/augmentation.py#L317) that randomly drops out between zero and all but one channel in the input. As we aimed to stick closely to the original trained setup of the different models, we did not use ChannelDropout for the other models. In principle, it would be a one-line code change in our [benchmark code](https://github.com/seisbench/pick-benchmark) to train the other models with this augmentation as well. Regarding transfer learning, this would definitely work as well. I'm just unsure if it is actually beneficial to train a model that can only process 1C data, if building a model to cope with missing channels is similarly easy as well.

So in short, EQTransformer with `strict=False` should do the job and you might also be lucky with the other models with the same parameter setting.

**Comment by dttrugman** [comment]
Thank you for the detailed information! I missed this feature entirely. Great to know, and thanks again for your time!

Daniel Trugman (he/him/his)
Assistant Professor
Nevada Seismological Laboratory
University of Nevada, Reno

________________________________
